assets/js/modules/analytics/datastore/properties.js

```diff
@@ -173,21 +173,24 @@ const baseActions = {
 
 			registry.dispatch( STORE_NAME ).setInternalWebPropertyID( internalPropertyID || '' );
 
-			// Clear any profile ID selection in the case that selection falls to the getProfiles resolver.
-			registry.dispatch( STORE_NAME ).setProfileID( '' );
+			const existingProfileID = registry.select( STORE_NAME ).getProfileID(); // eslint-disable-line @wordpress/no-unused-vars-before-return
+			const profiles = yield Data.commonActions.await( registry.__experimentalResolveSelect( STORE_NAME ).getProfiles( accountID, propertyID ) );
 
-			const profiles = registry.select( STORE_NAME ).getProfiles( accountID, propertyID );
-			if ( property.defaultProfileId && profiles?.some( ( profile ) => profile.id === property.defaultProfileId ) ) { // eslint-disable-line sitekit/acronym-case
-				registry.dispatch( STORE_NAME ).setProfileID( property.defaultProfileId ); // eslint-disable-line sitekit/acronym-case
-				return;
+			if ( profiles === undefined ) {
+				return; // Selection will happen in getProfiles resolver.
 			}
 
-			if ( profiles === undefined ) {
-				return; // Selection will happen in in getProfiles resolver.
+			let returnProfile = profiles.find( ( { id: ID } ) => ID === existingProfileID );
+
+			if ( ! returnProfile ) {
+				if ( property.defaultProfileId && profiles?.some( ( profile ) => profile.id === property.defaultProfileId ) ) { // eslint-disable-line sitekit/acronym-case
+					registry.dispatch( STORE_NAME ).setProfileID( property.defaultProfileId ); // eslint-disable-line sitekit/acronym-case
+					return;
+				}
+				returnProfile = profiles.find( ( { webPropertyId } ) => webPropertyId === propertyID ) || { id: PROFILE_CREATE }; // eslint-disable-line sitekit/acronym-case
```

Comment:
I realize this `find` was here from before it looks like it's my own left over from when another action was consolidated here that relied the `propertyID` alone. See https://github.com/google/site-kit-wp/commit/28fe5230a8cfd3c8973dd3f3bcb1b6ad6e61be6e

This can be simplified to get the first profile in the list instead (or fall back to create as it does) since all profiles in `profiles` should be for the given `propertyID`.

Reply:
Updated in https://github.com/google/site-kit-wp/pull/3180/commits/27342ee6405bbea9a567439024d735214f4ca1c2

Reply:
updated in https://github.com/google/site-kit-wp/pull/3180/commits/27342ee6405bbea9a567439024d735214f4ca1c2